miio/gateway.py

```diff
@@ -41,55 +41,24 @@ class DeviceType(IntEnum):
     AqaraMagnet = 53
 
 
-class SubDevice:
-    def __init__(self, gw, sid, type_, _, __, ___):
-        self.gw = gw
-        self.sid = sid
-        self.type = DeviceType(type_)
-
-    def unpair(self):
-        return self.gw.send("remove_device", [self.sid])
-
-    def battery(self):
-        return self.gw.send("get_battery", [self.sid])[0]
-
-    def get_firmware_version(self) -> Optional[int]:
-        """Returns firmware version"""
-        try:
-            return self.gw.send("get_device_prop", [self.sid, "fw_ver"])[0]
-        except Exception as ex:
-            _LOGGER.debug("Got an exception while fetching fw_ver: %s", ex, exc_info=True)
-            return None
-
-    def __repr__(self):
-        return "<Subdevice %s: %s fw: %s bat: %s>" % (self.type, self.sid, self.get_firmware_version(), self.battery())
-
-
-class SensorHT(SubDevice):
-    accessor = "get_prop_sensor_ht"
-    properties = ["temperature", "humidity"]
-
+class Gateway(Device):
+    """Main class representing the Xiaomi Gateway."""
 
-class Plug(SubDevice):
-    accessor = "get_prop_plug"
-    properties = ["power", "neutral_0"]
+    def __init__(self, Device) -> None:
+        self._device = Device
+        self._alarm = GatewayAlarm(Device)
```

Comment:
You should pass `self` instead of `Device` class reference here.

Reply:
Ah I see, thanks for pointing that out, it is now a lot nicer, besides I now improved the init function so you don't have to create a device first and then pass that but can just create the gateway class at once:

old:
```
from miio import (Device, gateway)
miio_device = Device("192.168.1.IP", 'token')
gateway = gateway.Gateway(miio_device)
gateway.alarm.on()
```

new:
```
from miio import (gateway)
gateway = gateway.Gateway("192.168.1.IP", 'token')
gateway.alarm.on()
```